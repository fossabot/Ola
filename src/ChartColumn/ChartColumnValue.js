import React from 'react'
import {default as PT} from 'prop-types'

const ChartColumnValue = ({ value, footer, children,  }) => {
  return (
    <div className="ola_chartColumn-value" style={{ '--value': value }}>
      <div className="ola_chartColumn-label">{ children }</div>
      <div className="ola_chartColumn-graphic"></div>
      { footer && <div className="ola_chartColumn-footer">{ footer }</div> }
    </div>
  )
}

ChartColumnValue.defaultProps = {
  value: 0
}

ChartColumnValue.propTypes = {
  /** Value between 0 and 1 */
  value: PT.number,
  /** Footer nodes */
  footer: PT.oneOfType([
    PT.arrayOf(PT.node),
    PT.node
  ]),
  /** Childen nodes */
  children: PT.oneOfType([
    PT.arrayOf(PT.node),
    PT.node
  ])
}

export default ChartColumnValue