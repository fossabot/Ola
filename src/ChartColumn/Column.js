import React from 'react'
import {default as PT} from 'prop-types'
import cx from 'classnames'

const Column = ({ value, footer, children, className, color }) => {
  const styles = cx('ola_chartColumn-value', className)
  return (
    <div className={styles} style={{ '--value': value, '--color': color }}>
      <div className="ola_chartColumn-label">{ children }</div>
      <div className="ola_chartColumn-graphic"></div>
      { footer && <div className="ola_chartColumn-footer">{ footer }</div> }
    </div>
  )
}

Column.defaultProps = {
  value: 0,
  className: null,
  color: null
}

Column.propTypes = {
  /** Value between 0 and 1 */
  value: PT.number,
  /** Extra className */
  className: PT.string,
  /** Color of the column */
  color: PT.string,
  /** Footer nodes */
  footer: PT.oneOfType([
    PT.arrayOf(PT.node),
    PT.node
  ]),
  /** Childen nodes */
  children: PT.oneOfType([
    PT.arrayOf(PT.node),
    PT.node
  ])
}

export default Column