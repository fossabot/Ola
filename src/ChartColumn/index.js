import React from 'react'
import {default as PT} from 'prop-types'
import cx from 'classnames'

const ChartColumn = ({ children, className, rule }) => {
  const styles = cx('ola_chartColumn', className)

  return (
    <div className={styles}>
      { children }
      { rule && <span className="ola_chartColumn-rule" style={{ '--value': rule.value }}>{ rule.name} </span> }
    </div>
  )
}

ChartColumn.defaultProps = {
  className: null,
  rule: null,
}

ChartColumn.propTypes = {
  /** Childen nodes */
  children: PT.oneOfType([
    PT.arrayOf(PT.node),
    PT.node
  ]).isRequired,
  /** Extra className */
  className: PT.string,
  /** Horizontal rule */
  rule: PT.exact({
    name: PT.string,
    value: PT.number,
  })
}

export default ChartColumn
