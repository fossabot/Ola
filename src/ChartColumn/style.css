.ola_chartColumn {
    --height: 150px;
    --color: var(--gray-light);

    display: flex;
    flex-direction: column;
}

.ola_chartColumn-value {
    display: flex;
    flex-direction: column;
    justify-content: flex-end;
    flex-grow: 1;
}

.ola_chartColumn-graphic {
    color: var(--color);
    background-color: var(--color);
    border-bottom: solid 1px;
    border-left: solid 1px;
}

.ola_chartColumn-footer {
    padding-top: var(--size-2);
    font: var(--callout);
    font-weight: var(--bold);
    color: var(--gray);
}
.ola_chartColumn-label {
    padding-bottom: var(--size-2);
    font: var(--callout);
}

@media (--to-width-1) {
    .ola_chartColumn-value + .ola_chartColumn-value {
        margin-top: var(--size-7);
    }

    .ola_chartColumn-graphic {
        width: calc(100% * var(--value));
        height: var(--size-5);
        border-top-right-radius: var(--radius-small);
        border-bottom-right-radius: var(--radius-small);
    }
}

@media (--from-width-1) {
    .ola_chartColumn {
        flex-direction: row;
        text-align: center;
    }

    .ola_chartColumn-footer {
        border-top: solid 1px var(--gray);
    }

    .ola_chartColumn-graphic {
        height: calc(var(--height) * var(--value));
        border-top-right-radius: var(--radius-small);
        border-top-left-radius: var(--radius-small);
        margin-left: var(--size-3);
        margin-right: var(--size-3);
    }
}