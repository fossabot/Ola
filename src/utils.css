/* Deprecated, use .ola-bg-gray-xlight instead */
.ola-bg {
    background-color: var(--gray-xlight);
}

/* Text sizes */
.ola-display {
    font: var(--font-display);
    letter-spacing: var(--font-display-spacing);
}
.ola-title {
    font: var(--font-title);
    letter-spacing: var(--font-title-spacing);
}
.ola-headline {
    font: var(--font-headline);
}
.ola-body {
    font: var(--font-body);
}
.ola-callout {
    font: var(--font-callout);
}
.ola-caption {
    font: var(--font-caption);
}

/* Colors */
.ola-brand {
    color: var(--brand);
}
.ola-white {
    color: var(--white);
}
.ola-black {
    color: var(--black);
}
.ola-gray {
    color: var(--gray);
}
.ola-gray-light {
    color: var(--gray-light);
}
.ola-gray-xlight {
    color: var(--gray-xlight);
}
.ola-error {
    color: var(--error);
}
.ola-warning {
    color: var(--warning);
}
.ola-success {
    color: var(--success);
}
.ola-pro {
    color: var(--pro);
}

/* Background colors */
.ola-bg-brand {
    background-color: var(--brand);
}
.ola-bg-white {
    background-color: var(--white);
}
.ola-bg-black {
    background-color: var(--black);
}
.ola-bg-gray {
    background-color: var(--gray);
}
.ola-bg-gray-light {
    background-color: var(--gray-light);
}
.ola-bg-gray-xlight {
    background-color: var(--gray-xlight);
}
.ola-bg-error {
    background-color: var(--error);
}
.ola-bg-warning {
    background-color: var(--warning);
}
.ola-bg-success {
    background-color: var(--success);
}
.ola-bg-pro {
    background-color: var(--pro);
}

/* Alignment */
.ola-right {
    text-align: right;
}
.ola-center {
    text-align: center;
}
.ola-numeric {
    font-variant-numeric: tabular-nums;
    text-align: right;
}

/* Misc */
.ola-nomargin {
    margin: 0;
}
.ola-bold {
    font-weight: var(--bold);
}