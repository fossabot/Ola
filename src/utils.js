/**
 * getElementType
 *
 * Returns a createElement() type based on the props of the Component.
 * Useful for calculating what type a component should render as.
 * @param {function} Component A function or ReactClass.
 * @param {object} props A ReactElement props object
 * @returns {string|function} A ReactElement type
*/

const getElementType = (Component, props) => {
  const { defaultProps = {} } = Component
  if (props.as && props.as !== defaultProps.as) return props.as
  if (props.href) return 'a'
  return defaultProps.as || 'div'
}

export { getElementType }




/**
 * normalizeRange
 *
 * Returns a number with a normalizade range based on the max and min value.
 * Useful for calculating ChartColumnValue values where is a number between 0 and 1.
 * @param {val} Value of current element
 * @param {max} Max value of all elements 
 * @param {min} Min value of all elements 
 * @returns {number} 
*/
const normalizeRange = (val, max, min = 0) => (val - min) / (max - min)
export { normalizeRange }

export function getRanges(values) {
  const max = Math.max(...values)
  return values.map(value => normalizeRange(value, max))
}