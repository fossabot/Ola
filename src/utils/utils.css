/* Deprecated, use .ola-bg-gray-xlight instead */
.ola-bg {
    background-color: var(--gray-xlight);
}

/* Text sizes */
.ola-display {
    font: var(--font-display);
    letter-spacing: var(--font-display-spacing);
}
.ola-title {
    font: var(--font-title);
    letter-spacing: var(--font-title-spacing);
}
.ola-headline {
    font: var(--font-headline);
}
.ola-body {
    font: var(--font-body);
}
.ola-callout {
    font: var(--font-callout);
}
.ola-caption {
    font: var(--font-caption);
}

h1,
h2,
h3,
h4,
h5,
h6,
b,
strong {
    &.ola-headline,
    &.ola-body,
    &.ola-callout,
    &.ola-caption {
        font-weight: var(--bold);
    }
}

/* Colors */
.ola-brand {
    color: var(--brand);
}
.ola-accent {
    color: var(--accent);
}
.ola-white {
    color: var(--white);
}
.ola-black {
    color: var(--black);
}
.ola-gray {
    color: var(--gray);
}
.ola-gray-light {
    color: var(--gray-light);
}
.ola-gray-xlight {
    color: var(--gray-xlight);
}
.ola-error {
    color: var(--error);
}
.ola-warning {
    color: var(--warning);
}
.ola-success {
    color: var(--success);
}
.ola-pro {
    color: var(--pro);
}
.ola-black-bolds b,
.ola-black-bolds strong {
    color: var(--black);
}

/* Background colors */
.ola-bg-brand {
    background-color: var(--brand);
}
.ola-bg-accent {
    background-color: var(--accent);
}
.ola-bg-white {
    background-color: var(--white);
}
.ola-bg-black {
    background-color: var(--black);
}
.ola-bg-gray {
    background-color: var(--gray);
}
.ola-bg-gray-light {
    background-color: var(--gray-light);
}
.ola-bg-gray-xlight {
    background-color: var(--gray-xlight);
}
.ola-bg-error {
    background-color: var(--error);
}
.ola-bg-warning {
    background-color: var(--warning);
}
.ola-bg-success {
    background-color: var(--success);
}
.ola-bg-pro {
    background-color: var(--pro);
}

/* Alignment */
.ola-right {
    text-align: right;
}
.ola-center {
    text-align: center;
}
.ola-numeric {
    font-variant-numeric: tabular-nums;
    text-align: right;
}

/* Misc */
.ola-nomargin {
    margin: 0;
}
.ola-nomargin-top {
    margin-top: 0;
}
.ola-nomargin-bottom {
    margin-bottom: 0;
}
.ola-bold {
    font-weight: var(--bold);
}