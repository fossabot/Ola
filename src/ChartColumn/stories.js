import React from 'react'
import ChartColumn from './'
import ChartColumnValue from './ChartColumnValue'
import { normalizeRange } from '../utils'

const mockData = [
  {
    value: 5,
    label: '5 keywords'
  },
  {
    value: 2,
    label: '2 keywords'
  },
  {
    value: 1,
    label: '1 keywords'
  },
  {
    value: 10,
    label: '10 keywords'
  }
]

export default {
  title: 'ChartColumn',
  component: ChartColumn,
  args: {}
}

export const Base = () => {
  const extractValues = mockData.map( data => data.value )
  return (
    <ChartColumn>
      { mockData.map((row, idx) => {
        const normalizedValue = normalizeRange(row.value, Math.max(...extractValues), Math.min(...extractValues))
        return (
          <ChartColumnValue key={idx} value={normalizedValue}>{row.label}</ChartColumnValue>
        )
      })}
    </ChartColumn>
  )
}
 
export const HtmlLabels = () => {
  const extractValues = mockData.map( data => data.value )
  return (
    <ChartColumn>
      { mockData.map((row, idx) => {
        const normalizedValue = normalizeRange(row.value, Math.max(...extractValues), Math.min(...extractValues))
        return (
          <ChartColumnValue key={idx} value={normalizedValue}><div><p>{row.label}</p><p>{row.label}</p></div></ChartColumnValue>
        )
      })}
    </ChartColumn>
  )
}

export const ColumnsWithFooter = () => {
  const extractValues = mockData.map( data => data.value )
  return (
    <ChartColumn>
      { mockData.map((row, idx) => {
        const normalizedValue = normalizeRange(row.value, Math.max(...extractValues), Math.min(...extractValues))
        const footer = ["one", "two", "tree", "four"]
        return (
          <ChartColumnValue 
            key={idx}
            footer={footer[idx]}
            value={normalizedValue}>
              <div><p>{row.label}</p><p>{row.label}</p></div>
          </ChartColumnValue>
        )
      })}
    </ChartColumn>
  )
}