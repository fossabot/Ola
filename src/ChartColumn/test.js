import React from 'react'
import ChartColumn from './'
import ChartColumnValue from './ChartColumnValue'
import { normalizeRange } from '../utils'
import renderer from 'react-test-renderer'

const mockData = [
  {
    value: 5,
    label: '5 keywords'
  },
  {
    value: 2,
    label: '2 keywords'
  },
  {
    value: 1,
    label: '1 keywords'
  },
  {
    value: 10,
    label: '10 keywords'
  }
]

it('Default ChartColumn', () => {
  const tree = renderer
    .create(<ChartColumn>Test</ChartColumn>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('Default ChartColumnValue', () => {
  const tree = renderer
    .create(<ChartColumnValue></ChartColumnValue>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('ChartColumn and ChartColumnValue', () => {
  const extractValues = mockData.map( data => data.value )
  const tree = renderer
    .create(
      <ChartColumn>
        { mockData.map((row, idx) => {
          const normalizedValue = normalizeRange(row.value, Math.max(...extractValues), Math.min(...extractValues))
          return (
            <ChartColumnValue key={idx} value={normalizedValue}>{row.label}</ChartColumnValue>
          )
        })}
      </ChartColumn>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})


