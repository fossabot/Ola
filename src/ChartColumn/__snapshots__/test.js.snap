// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChartColumn and ChartColumnValue 1`] = `
<div
  className="ola_chartColumn"
>
  <div
    className="ola_chartColumn-value"
    style={
      Object {
        "--color": null,
        "--value": 0.4444444444444444,
      }
    }
  >
    <div
      className="ola_chartColumn-label"
    >
      5 keywords
    </div>
    <div
      className="ola_chartColumn-graphic"
    />
  </div>
  <div
    className="ola_chartColumn-value"
    style={
      Object {
        "--color": null,
        "--value": 0.1111111111111111,
      }
    }
  >
    <div
      className="ola_chartColumn-label"
    >
      2 keywords
    </div>
    <div
      className="ola_chartColumn-graphic"
    />
  </div>
  <div
    className="ola_chartColumn-value"
    style={
      Object {
        "--color": null,
        "--value": 0,
      }
    }
  >
    <div
      className="ola_chartColumn-label"
    >
      1 keywords
    </div>
    <div
      className="ola_chartColumn-graphic"
    />
  </div>
  <div
    className="ola_chartColumn-value"
    style={
      Object {
        "--color": null,
        "--value": 1,
      }
    }
  >
    <div
      className="ola_chartColumn-label"
    >
      10 keywords
    </div>
    <div
      className="ola_chartColumn-graphic"
    />
  </div>
</div>
`;

exports[`Default ChartColumn 1`] = `
<div
  className="ola_chartColumn"
>
  Test
</div>
`;

exports[`Default ChartColumnValue 1`] = `
<div
  className="ola_chartColumn-value"
  style={
    Object {
      "--color": null,
      "--value": 0,
    }
  }
>
  <div
    className="ola_chartColumn-label"
  />
  <div
    className="ola_chartColumn-graphic"
  />
</div>
`;
